### ISSUE TITLE
Add support for operation name

### ISSUE THREAD

**Issue by chevinbrown** [dillonkearns/elm-graphql#123]
We use the `OperationName` for "tagging" queries so we can separate them in our APM. 
Example: 

<img width="638" alt="Screen Shot 2019-04-29 at 10 18 51 AM" src="https://user-images.githubusercontent.com/864581/56906611-3764b280-6a68-11e9-926f-fe4d26878261.png">

This helps us diagnose performance issues in particular query-groups. Since we can't do that in elm, they all get lumped into our "web requests" as a catch-all, so there's no way to distinguish real queries versus types that wrap APIs, for example. 

Would it be possible for elm-graphql to do this? 

Thanks for all your work on this library!

**Comment by dillonkearns** [comment]
Hey Kevin, thanks for the context, that's super helpful!

Here's an idea for the API design.

```elm
query : SelectionSet String RootQuery
query =
    Query.hello

makeRequest : Cmd Msg
makeRequest =
    query
        |> Graphql.Http.queryRequest "https://elm-graphql.herokuapp.com"
        |> Graphql.Http.withOperationName "SayHello"
        |> Graphql.Http.send (RemoteData.fromResult >> GotResponse)
```

I like this design because it is simple and doesn't introduce any breaking changes or any complicated types that people will have to learn. But if you wanted to do something more sophisticated within a particular codebase, you could always make some helpers for yourself to enforce that queries are always named, for example:

```elm
module Request exposing (Response, query, OperationName(..))

import Graphql.Http
import Graphql.Operation exposing (RootQuery)
import Graphql.SelectionSet as SelectionSet exposing (SelectionSet)
import Http
import RemoteData exposing (RemoteData)

type alias Response data =
    RemoteData (Graphql.Http.RawError () Http.Error) data

type OperationName = OperationName String

query :
    (Response decodesTo -> msg)
    -> (OperationName, SelectionSet decodesTo RootQuery)
    -> Cmd msg
query msgConstructor (OperationName operationName, querySelection) =
    querySelection
        |> Graphql.Http.queryRequest "https://myapi.com"
        |> Graphql.Http.withOperationName operationName
        |> Graphql.Http.send
            (\result ->
                result
                    |> Graphql.Http.discardParsedErrorData
                    |> Graphql.Http.withSimpleHttpError
                    |> RemoteData.fromResult
                    |> msgConstructor
            )
```

Do you think that design would solve your problem well?

**Comment by chevinbrown** [comment]
💯 that looks fantastic to me!

**Comment by asterite** [comment]
@dillonkearns Hi! Is it OK if I try to submit a PR for this feature with the API you proposed?

```elm
query : SelectionSet String RootQuery
query =
    Query.hello

makeRequest : Cmd Msg
makeRequest =
    query
        |> Graphql.Http.queryRequest "https://elm-graphql.herokuapp.com"
        |> Graphql.Http.withOperationName "SayHello"
        |> Graphql.Http.send (RemoteData.fromResult >> GotResponse)
```

**Comment by pellebirkstrom** [comment]
It would be awesome to have this merged!

**Comment by dillonkearns** [comment]
I merged and released the PR for this, this is now available in Elm package version 4.4.0 👍  https://github.com/dillonkearns/elm-graphql/blob/master/CHANGELOG-ELM-PACKAGE.md#440---2019-10-14
